### PR TITLE
Removes the need to make handholes for jumpsuit sprites

### PR DESCRIPTION
**Pull request by ElorgRHG**
## What Does This PR Do

Hands now layer on top of jumpsuits, but below gloves, no actual player-side changes to be seen.

## Why It's Good For The Game

Makes spriting, or porting sprites, easier due to not having to cut random holes in your sprites.
Makes it also a bit easier to sprite for species.

## Testing

Took a jumpsuit sprite.
Filled the handhole with a solid color.
Checked ingame.
Hand overlayed over solid color filled hole.

## Changelog

Nothing player facing hopefully.

### CODE REVIEW

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -165,7 +165,6 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 
 //BASE MOB SPRITE
 /mob/living/carbon/human/proc/update_body(rebuild_base = FALSE)
-	remove_overlay(BODY_LAYER)
```
Comment (ElorgRHG) [inline code/modules/mob/living/carbon/human/update_icons.dm:168]: `BODY_LAYER` wasn't used anywhere as far as I know so I just removed it.

**Comment by russetdapone** [comment]
Hurry and merge it so I can port all the TG outfits

**Comment by McRamon** [comment]
This is a must

**Comment by Christasmurf** [comment]
Does this affect suits (jackets, hardsuits) as well or is this only jumpsuits/uniforms?

**Comment by ElorgRHG** [comment]
> Does this affect suits (jackets, hardsuits) as well or is this only jumpsuits/uniforms?

_**Only jumpsuits**_, some suits have inbuilt "gloves", hardsuits being the best example, so it would look weird having just a hand stick out of some.

**Comment by ElorgRHG** [comment]
Found some small issues, no time to fix right now, draft till then.

**Comment by ElorgRHG** [comment]
Hopefully this fixes more than it breaks

File: code/modules/client/preference/character.dm [hunk 2]
```diff
@@ -822,6 +822,9 @@
 		else
 			preview_icon.Blend(rgb(-s_tone,  -s_tone,  -s_tone), ICON_SUBTRACT)
 
+	var/icon/hands_icon = icon(preview_icon)
+	hands_icon.Blend(icon('icons/mob/clothing/masking_helpers.dmi', "hands_mask"), ICON_MULTIPLY)
+
```
Comment (SteelSlayer) [inline code/modules/client/preference/character.dm:827]: Could you please explain what the mask is doing here? 

Also the masking helper sprite looks like it's for a human body only, is that going to be a problem when applied to other species?

Comment (ElorgRHG) [reply code/modules/client/preference/character.dm:827]: To my dismay when I was making this PR it turned out the lobby character preview _renders_ in a completely different way than ingame, this mask exists to cut out a handhole in the lobby character preview by "force".

It doesnt work on species, but neither does the entire lobby character preview, it puts human sprites on all species, and changing that to work with species sprites would be a particularly large undertaking compared to the current size of this PR, and not within the scope.

**Review by SteelSlayer — APPROVED** [review]
Seems fine to me.

**Pull request merged**